src/main/java/com/cloudbees/jenkins/plugins/sshagent/RemoteAgent.java

```diff
@@ -43,12 +45,15 @@
      * @param privateKey the private key.
      * @param passphrase the passphrase or {@code null}.
      * @param comment    the comment to give to the key.
+     * @param launcher the launcher for the remote node.
      * @throws java.io.IOException if something went wrong.
      */
-    void addIdentity(String privateKey, String passphrase, String comment) throws IOException, InterruptedException;
+    void addIdentity(String privateKey, String passphrase, String comment, Launcher launcher) throws IOException, InterruptedException;
```

Comment:
Incompatible but I am inclined to let it pass given that [there are no _known_ external implementations](https://jenkins.io/doc/developer/extensions/ssh-agent/).

Reply:
My thought as well, not publicly used elsewhere and very unlikely that anyone have local implementations of this.